internal/controller/postgrescluster/util_test.go

```diff
@@ -325,42 +341,44 @@ func TestAddNSSWrapper(t *testing.T) {
 
 			addNSSWrapper(image, imagePullPolicy, template)
 
-			// verify proper nss_wrapper env vars
-			var expectedContainerUpdateCount int
-			for i, c := range template.Spec.Containers {
-				if c.Name == "database" || c.Name == "pgbackrest" {
-					assert.DeepEqual(t, expectedEnv, c.Env)
-					expectedContainerUpdateCount++
-				} else {
-					assert.DeepEqual(t, beforeAddNSS[i], c)
+			t.Run("container-updated", func(t *testing.T) {
+				// Each container that requires the nss_wrapper envs should be updated
+				var actualUpdatedContainerCount int
+				for i, c := range template.Spec.Containers {
+					if c.Name == naming.ContainerDatabase ||
+						c.Name == naming.PGBackRestRepoContainerName ||
+						c.Name == naming.ContainerPGUpgrade ||
+						c.Name == naming.PGBackRestRestoreContainerName {
+						assert.DeepEqual(t, expectedEnv, c.Env)
+						actualUpdatedContainerCount++
+					} else {
+						assert.DeepEqual(t, beforeAddNSS[i], c)
+					}
 				}
-			}
 
-			// verify database and/or pgbackrest containers updated
-			assert.Equal(t, expectedContainerUpdateCount,
-				databaseBackrestContainerCount(template))
-
-			var foundInitContainer bool
-			// verify init container command, image & name
-			for _, c := range template.Spec.InitContainers {
-				if c.Name == naming.ContainerNSSWrapperInit {
-					assert.Equal(t, expectedCmd, c.Command[2]) // ignore "bash -c"
-					assert.Assert(t, c.Image == image)
-					assert.Assert(t, c.ImagePullPolicy == imagePullPolicy)
-					assert.Assert(t, c.SecurityContext != &corev1.SecurityContext{})
-
-					for i, c := range template.Spec.Containers {
-						if c.Name == "database" || c.Name == "pgbackrest" {
-							assert.DeepEqual(t, c.Resources.Requests,
-								template.Spec.Containers[i].Resources.Requests)
-						}
+				// verify database and/or pgbackrest containers updated
+				assert.Equal(t, actualUpdatedContainerCount,
+					expectedUpdatedContainerCount(template))
```

Comment:
🏕️ This would make more sense to me as a property of the test case. Is there something that prevents that?

```go
	tcName: "database container with pgbackrest sidecar",
	expected: 2,
	podTemplate: &corev1.PodTemplateSpec{Spec: corev1.PodSpec{
	…
```

```go
	// verify database and/or pgbackrest containers updated
	assert.Equal(t, actualUpdatedContainerCount, tc.expected)
```